test/functional/command_write_concern.test.js

```diff
@@ -17,245 +17,160 @@ var extend = function(template, fields) {
-      var firstSecondary = [
-        extend(defaultFields, {
-          ismaster: false,
-          secondary: true,
-          me: 'localhost:32001',
-          primary: 'localhost:32000',
-          tags: { loc: 'sf' }
-        })
-      ];
-
-      // Primary server states
-      var arbiter = [
-        extend(defaultFields, {
-          ismaster: false,
-          secondary: false,
-          arbiterOnly: true,
-          me: 'localhost:32002',
-          primary: 'localhost:32000'
-        })
-      ];
-
-      // Boot the mock
-      co(function*() {
-        const primaryServer = yield mock.createServer(32000, 'localhost');
-        const firstSecondaryServer = yield mock.createServer(32001, 'localhost');
-        const arbiterServer = yield mock.createServer(32002, 'localhost');
-
-        primaryServer.setMessageHandler(request => {
-          var doc = request.document;
-
-          if (doc.ismaster) {
-            request.reply(primary[0]);
-          } else if (doc.listCollections) {
-            request.reply({
-              ok: 1,
-              cursor: {
-                id: Long.fromNumber(0),
-                ns: 'test.cmd$.listCollections',
-                firstBatch: []
-              }
-            });
-          } else if (doc.create) {
-            commandResult = doc;
-            request.reply({ ok: 1 });
-          } else if (doc.endSessions) {
-            request.reply({ ok: 1 });
-          }
-        });
-
-        firstSecondaryServer.setMessageHandler(request => {
-          var doc = request.document;
-          if (doc.ismaster) {
-            request.reply(firstSecondary[0]);
-          } else if (doc.endSessions) {
-            request.reply({ ok: 1 });
-          }
-        });
-
-        arbiterServer.setMessageHandler(request => {
-          var doc = request.document;
-          if (doc.ismaster) {
-            request.reply(arbiter[0]);
-          } else if (doc.endSessions) {
-            request.reply({ ok: 1 });
-          }
-        });
-
-        var commandResult = null;
-
-        // Connect to the mocks
-        const client = configuration.newClient(
-          'mongodb://localhost:32000,localhost:32001,localhost:32002/test?replicaSet=rs'
-        );
-
-        client.connect(function(err, client) {
-          test.equal(null, err);
-          var db = client.db(configuration.db);
+    test: writeConcernTest('aggregate', function(db, writeConcernTestOptions, done) {
+      db.collection('test')
+        .aggregate(
+          [{ $match: {} }, { $out: 'readConcernCollectionAggregate1Output' }],
+          writeConcernTestOptions
+        )
+        .toArray(done);
+    })
+  });
 
-          db.createCollection('test_collection_methods', { w: 2, wtimeout: 1000 }, function(err) {
-            test.equal(null, err);
-            test.deepEqual({ w: 2, wtimeout: 1000 }, commandResult.writeConcern);
+  it('successfully pass through writeConcern to create command', {
+    metadata: {
```

Comment:
two things:
- let's not use `generators: true` anymore, it's meaningless since we only support versions of node where generators are supported
- we're only targeting a single topology here because its a mock test, I think you should use the simplified test form to run on all variants:
```
it('successfully pass through writeConcern to create command', function(done) {
  ... test ...
});
```